plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/ScriptImpl.java

```diff
@@ -28,67 +28,117 @@
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * ScriptImpl can be used as either an {@link ExecutableScript} or a {@link LeafSearchScript}
+ * to run a previously compiled Plan A script.
+ */
 final class ScriptImpl implements ExecutableScript, LeafSearchScript {
-    final Executable executable;
-    final Map<String,Object> variables;
-    final LeafSearchLookup lookup;
-    
-    ScriptImpl(Executable executable, Map<String,Object> vars, LeafSearchLookup lookup) {
+    private final Executable executable;
+    private final Map<String, Object> variables;
+    private final LeafSearchLookup lookup;
+
+    /**
+     * Creates a ScriptImpl for the a previously compiled Plan A script.
+     * @param executable The previously compiled Plan A script.
+     * @param vars The initial variables to run the script with.
+     * @param lookup The lookup to allow search fields to be available if this is run as a search script.
+     */
+    ScriptImpl(final Executable executable, final Map<String, Object> vars, final LeafSearchLookup lookup) {
         this.executable = executable;
         this.lookup = lookup;
         this.variables = new HashMap<>();
+
         if (vars != null) {
             variables.putAll(vars);
         }
+
         if (lookup != null) {
             variables.putAll(lookup.asMap());
         }
     }
-    
+
+    /**
+     * Set a variable for the script to be run against.
+     * @param name The variable name.
+     * @param value The variable value.
+     */
     @Override
-    public void setNextVar(String name, Object value) {
+    public void setNextVar(final String name, final Object value) {
         variables.put(name, value);
     }
-    
+
+    /**
+     * Run the script.
+     * @return The script result.
+     */
     @Override
     public Object run() {
         return executable.execute(variables);
     }
-    
+
+    /**
+     * Run the script.
+     * @return The script result as a double.
+     */
+    @Override
+    public double runAsDouble() {
+        return ((Number)run()).doubleValue();
+    }
+
+    /**
+     * Run the script.
+     * @return The script result as a float.
+     */
     @Override
     public float runAsFloat() {
-        return ((Number) run()).floatValue();
+        return ((Number)run()).floatValue();
     }
 
+    /**
+     * Run the script.
+     * @return The script result as a long.
+     */
     @Override
     public long runAsLong() {
-        return ((Number) run()).longValue();
+        return ((Number)run()).longValue();
     }
 
+    /**
+     * This method has no effect in Plan A.
+     * @param value The value to unwrap.
+     * @return The value passed in.
```

Comment:
It might be worth adding a default implementation of it to whatever interface declares it. The vast majority of the time its a noop.